### PR TITLE
Implementing #38

### PR DESCRIPTION
**Pull request by adamdriscoll**

### CODE REVIEW

File: PowerShellTools/PowerShellToolsPackage.cs [hunk 1]
```diff
@@ -211,6 +220,53 @@ protected override void Initialize()
...
+            get
+            {
+                var version = new Version(0, 0);
+                using (var reg = Registry.LocalMachine.OpenSubKey(@"Software\Microsoft\PowerShell\3\PowerShellEngine"))
+                {
+                    if (reg != null)
+                    {
+                        var versionString = reg.GetValue("PowerShellVersion") as string;
+
+                        Version.TryParse(versionString, out version);
```
Comment (AndreSayreMSFT) [inline PowerShellTools/PowerShellToolsPackage.cs]: if versionString is null or String.Empty, version will also be null when TryParse returns, and we return a null version.  Intentional?

File: PowerShellTools/PowerShellToolsPackage.cs [hunk 2]
```diff
@@ -211,6 +220,53 @@ protected override void Initialize()
...
+                        return version;
+                    }
+                }
+
+                using (var reg = Registry.LocalMachine.OpenSubKey(@"Software\Microsoft\PowerShell\1\PowerShellEngine"))
+                {
+                    if (reg != null)
+                    {
+                        var versionString = reg.GetValue("PowerShellVersion") as string;
+                        Version.TryParse(versionString, out version);
```
Comment (AndreSayreMSFT) [inline PowerShellTools/PowerShellToolsPackage.cs]: Same comment as above re: versionString being null

File: PowerShellTools/Project/PowerShellProjectPackage.cs [hunk 3]
```diff
@@ -13,6 +13,11 @@ namespace PowerShellTools.Project
     [ProvideEditorExtension(typeof(PowerShellEditorFactory), PowerShellConstants.PS1File, 50, ProjectGuid = VSConstants.CLSID.MiscellaneousFilesProject_string, NameResourceID = 3004, DefaultName = "module", TemplateDir = "NewItemTemplates")]
     public class PowerShellProjectPackage : CommonProjectPackage
     {
+        protected override void Initialize()
```
Comment (AndreSayreMSFT) [inline PowerShellTools/Project/PowerShellProjectPackage.cs]: Will ValidateInstalledPowerShellVersion() be called multiple times; Once per package initialization, resulting in multiple prompts to the user?

File: PowerShellTools/Project/PowerShellProjectPackage.cs [hunk 4]
```diff
@@ -13,6 +13,11 @@ namespace PowerShellTools.Project
     [ProvideEditorExtension(typeof(PowerShellEditorFactory), PowerShellConstants.PS1File, 50, ProjectGuid = VSConstants.CLSID.MiscellaneousFilesProject_string, NameResourceID = 3004, DefaultName = "module", TemplateDir = "NewItemTemplates")]
     public class PowerShellProjectPackage : CommonProjectPackage
```
Comment (AndreSayreMSFT) [inline PowerShellTools/Project/PowerShellProjectPackage.cs]: Do we need validation upon the REPL Window's package initialization?

File: PowerShellTools/PowerShellToolsPackage.cs (hunk 1 above)
Comment (adamdriscoll) [inline PowerShellTools/PowerShellToolsPackage.cs]: Probably should have it return 0.0.

File: PowerShellTools/Project/PowerShellProjectPackage.cs (hunk 3 above)
Comment (adamdriscoll) [inline PowerShellTools/Project/PowerShellProjectPackage.cs]: I have noticed to prompts once per package.

File: PowerShellTools/Project/PowerShellProjectPackage.cs (hunk 4 above)
Comment (adamdriscoll) [inline PowerShellTools/Project/PowerShellProjectPackage.cs]: Hmmm good point. We might. I'll have to test that.

**Comment by adamdriscoll** [comment]
I think I'm going to see if there is some way to do this once per VSIX rather than once per package.

**Pull request closed without merging**